vlib/v/gen/native/tests/libc.vv

```diff
@@ -1,38 +1,47 @@
 // because of an issue with checker, all C.* functions have to be declared first
 fn C.isalpha(c int) int
 fn C.isdigit(c int) int
+fn C.malloc(n u64) voidptr
+fn C.free(ptr voidptr)
+fn C.memset(ptr voidptr, c int, n u64) voidptr
 
 fn main() {
 	charutil()
+	memory()
 }
 
 fn charutil() {
+	// only linux supports linking right now
 	$if linux {
 		// ascii for `V`
 		v_is_alpha := C.isalpha(86)
-		if v_is_alpha != 0 {
-			println('ok 1')
-		} else {
-			assert false
-		}
+		assert v_is_alpha != 0
 
 		null_is_alpha := C.isalpha(0)
-		if null_is_alpha == 0 {
-			println('ok 2')
-		} else {
-			assert false
-		}
+		assert null_is_alpha == 0
 
 		// ascii for `3`
 		three_is_digit := C.isdigit(51)
-		if three_is_digit != 0 {
-			println('ok 3')
-		} else {
-			assert false
-		}
-	} $else {
-		println('ok 1')
-		println('ok 2')
-		println('ok 3')
+		assert three_is_digit != 0
 	}
 }
+
+fn memory() {
+	$if linux {
+		// allocate some memory
+		buf := &int(C.malloc(100))
+		assert buf != unsafe { nil }
+
+		C.memset(buf, 0, 100)
+		assert *buf == 0
+
+		unsafe {
+			*buf = 123456
+			*(buf + sizeof(int)) = 100
```

Comment:
`buf + 1` should be equivalent to `&int( u64(buf) + sizeof(int) )` .

Reply:
Yea, noticed that too. I'll have to fix that